cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationManager.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.plugins;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closeables;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manage locations for {@link RollingLocationLogAppender}
+ */
+public class LocationManager {
+  protected static final String TAG_NAMESPACE_ID = ".namespaceId";
+  protected static final String TAG_APPLICATION_ID = ".applicationId";
+
+  private Location logDirLocation;
+
+  private Map<LocationIdentifier, OutputStream> activeFiles;
+  private Map<LocationIdentifier, Location> activeFilesToLocation;
+
+  public LocationManager(LocationFactory locationFactory) {
+    this.logDirLocation = locationFactory.create("plugins/logs");
+    this.activeFiles = new HashMap<>();
+    this.activeFilesToLocation = new HashMap<>();
```

Comment:
Since each `Location#getOutputStream` creates a new map, we are using 2 maps. Is there a way to get Location out of OutputStream?